### PR TITLE
Creation of signing keys in setup appliance sh

### PR DESCRIPTION
**Pull request by M0ses**
* fixed ordering of daemon startup to grant bs_srcserver/bs_repserver are running when "rake writconfiguration" is done
* moved code for creating key for obssigner to setup-appliance.sh
* fixed creation of directories in rep and src server
* fixed testing

### CODE REVIEW

**Comment by M0ses** [comment]
@adrianschroeter @mlschroe @shyukri : Please review

File: dist/t/spec/features/0040_prepare_package_building_spec.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    find('input[name="commit"]').click #Save changes
+    expect(page).to have_content("Package 'obs-build' was created successfully")
+
+  end
+
+  # prepare for upload
+  download_url = "https://api.opensuse.org/public/source/OBS:Server:2.7/build"
+  dir = Dir.mktmpdir
+  # get spec file
+  upload_files = ['build.spec','_service','obs-build-20160426.tar.gz']
```
Comment (shyukri) [inline dist/t/spec/features/0040_prepare_package_building_spec.rb]: Are we sure that the archive won't change it's name/timestamp?

File: dist/obsapisetup [hunk 2]
```diff
@@ -10,9 +10,9 @@
 ### BEGIN INIT INFO
 # Provides:          obsapisetup
 # X-Start-Before:    apache2
-# Should-Start:      obsstoragesetup obssrcserver
+# Should-Start:      obsstoragesetup
 # Should-Stop:       $none
-# Required-Start:    mysql
+# Required-Start:    mysql obssrcserver obsrepserver
```
Comment (adrianschroeter) [inline dist/obsapisetup]: src and rep server may run on a different host. Okay, not the standard situation on the appliance, but "Should-Start" would be the better place.

File: dist/setup-appliance.sh [hunk 3]
```diff
@@ -586,8 +587,15 @@ if [[ ! $BOOTSTRAP_TEST_MODE == 1 && $0 != "-bash" ]];then
   # prepare configuration for obssigner before any other backend service
   # is started, because obssigner configuration might affect other services
   # too
+  GPG_KEY_CREATED=0
+
   prepare_obssigner
 
+  if [[ $GPG_KEY_CREATED == 1 ]];then
+    /usr/lib/obs/server/bs_srcserver --restart
+    /usr/lib/obs/server/bs_repserver --restart
```
Comment (adrianschroeter) [inline dist/setup-appliance.sh]: can you use the init script and use "reload" here instead?

It would not start them when they are not running.

**Pull request merged**